src/macros.rs

```diff
@@ -190,6 +190,22 @@ macro_rules! redis_module {
             use $crate::configuration::get_bool_default_config_value;
             use $crate::configuration::get_enum_default_config_value;
 
+            if ctx.is_null() {
+                return raw::Status::Err as _;
+            }
+
+            if argv.is_null() && argc != 0 {
+                return raw::Status::Err as _;
+            }
+
+            if !argv.is_null() && unsafe { (*argv).is_null() } {
```

Comment:
You assume here that none NULL  argv means that `argc > 0`. I am not sure this is promise.

Reply:
No, we should not get a negative value.